nucliadb/nucliadb/tests/integration/test_vectorsets.py

```
@@ -0,0 +1,112 @@
+from nucliadb.tests.utils import inject_message
+from nucliadb_protos import nodereader_pb2, writer_pb2
+
+DEFAULT_VECTOR_DIMENSION = 512
+VECTORSET_DIMENSION = 12
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("knowledgebox", ("EXPERIMENTAL", "STABLE"), indirect=True)
+async def test_vectorsets(
+    nucliadb_reader: AsyncClient,
+    nucliadb_writer: AsyncClient,
+    nucliadb_grpc: WriterStub,
+    knowledgebox: str,
+):
+    kbid = knowledgebox
+    vectorset_id = "my-vectorset"
+
+    await create_vectorset(nucliadb_grpc, kbid, vectorset_id)
+    await inject_broker_message_with_vectorset_data(nucliadb_grpc, kbid, vectorset_id)
+
+    # TODO: change this to use search REST API when implemented
+    shards = await manager.KBShardManager().get_shards_by_kbid(kbid)
+    logic_shard = shards[0]
+    node, shard_id = manager.choose_node(logic_shard)
+
+    test_cases = [
+        (DEFAULT_VECTOR_DIMENSION, ""),
+        (VECTORSET_DIMENSION, vectorset_id),
+    ]
+    for dimension, vectorset in test_cases:
+        query_pb = nodereader_pb2.SearchRequest(
+            shard=shard_id,
+            body="this is a query for my vectorset",
+            vector=[1.23] * dimension,
+            vectorset=vectorset,
+            result_per_page=5,
+        )
+        results = await node.reader.Search(query_pb)  # type: ignore
+        assert len(results.vector.documents) == 5
+
+    test_cases = [
+        (DEFAULT_VECTOR_DIMENSION, vectorset_id),
+        (VECTORSET_DIMENSION, ""),
+    ]
+    for dimension, vectorset in test_cases:
+        query_pb = nodereader_pb2.SearchRequest(
+            shard=shard_id,
+            body="this is a query for my vectorset",
+            vector=[1.23] * dimension,
+            vectorset=vectorset,
+            result_per_page=5,
+        )
+        with pytest.raises(Exception) as exc:
+            results = await node.reader.Search(query_pb)  # type: ignore
+            assert "inconsistent dimensions" in str(exc).lower()
```

I think you're swallowing any exceptions here and the assert is not happening. You need to de-indent the assert one level

---

Oh true! Thanks!